### PR TITLE
Remove vanilla pickle serialization for task arguments

### PR DESCRIPTION
**Pull request by edoakes**
## Why are these changes needed?

Now that we use pickle5, this is no longer necessary. Additionally, it's required for reference counting to collect `ObjectIDs` contained within objects during serialization so we need to use the custom serializer registered in `serialization.py`.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://ray.readthedocs.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.

### CODE REVIEW

File: python/ray/_raylet.pyx [hunk 1]
```diff
@@ -361,56 +280,30 @@ cdef c_vector[c_string] string_vector_from_list(list string_list):
...
 
-    # TODO be consistent with store_task_outputs
     for arg in args:
         if isinstance(arg, ObjectID):
             args_vector.push_back(
                 CTaskArg.PassByReference((<ObjectID>arg).native()))
-        elif not ray._raylet.check_simple_value(arg):
+        worker = ray.worker.global_worker
+        serialized_arg = worker.get_serialization_context().serialize(arg)
+        if serialized_arg.total_bytes > 100 * 1024:
```
Comment (edoakes) [inline python/ray/_raylet.pyx]: TODO: Will use the same constant that we use in core worker

Comment (pcmoritz) [reply python/ray/_raylet.pyx]: Sounds good!

File: python/ray/_raylet.pyx [hunk 2]
```diff
@@ -361,56 +280,30 @@ cdef c_vector[c_string] string_vector_from_list(list string_list):
...
             args_vector.push_back(
                 CTaskArg.PassByReference((<ObjectID>ray.put(arg)).native()))
-        elif type(arg) is bytes:
```
Comment (pcmoritz) [inline python/ray/_raylet.pyx]: Make sure this doesn't break the Java <-> Python argument passing

**Review by pcmoritz — APPROVED** [review]
LGTM if the comments are addressed and the tests pass :)

File: python/ray/_raylet.pyx (hunk 2 above)
Comment (edoakes) [reply python/ray/_raylet.pyx]: This is actually handled in `serialization.py` already

**Pull request merged**